### PR TITLE
Define HTTP User-Agent attribute

### PR DESCRIPTION
**Pull request by kbrockhoff**
Having user agent data is often useful in troubleshooting. There are also extreme use cases like my current project where only certain browsers are contractually supported. Storing with traces eliminates the need to jump back and forth between tracing backend and log aggregation backend.

### CODE REVIEW

**Comment by dyladan** [comment]
JS already does this https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-plugin-http/src/enums/AttributeNames.ts#L32

File: specification/data-http.md [hunk 1]
```diff
@@ -176,10 +176,12 @@ If the route cannot be determined, the `name` attribute MUST be set as defined i
 | `http.server_name` | The primary server name of the matched virtual host. This should be obtained via configuration. If no such configuration can be obtained, this attribute MUST NOT be set ( `net.host.name` should be used instead). | [1] |
 | `http.route` | The matched route (path template). (TODO: Define whether to prepend application root) E.g. `"/users/:userID?"`. | No |
 | `http.client_ip` | The IP address of the original client behind all proxies, if known (e.g. from [X-Forwarded-For][]). Note that this is not necessarily the same as `net.peer.ip`, which would identify the network-level peer, which may be a proxy. | No |
+| `http.user_agent` | Value of the HTTP [User-Agent] header sent by the client or else a calculated user agent device identifier compatible with the backend system in use. | No |
```
Comment (Oberon00) [inline specification/data-http.md]: What is a "calculated user agent device identifer"? I feel like we should avoid such vague wording here. Just specify the HTTP User-Agent header. If the backend and instrumentation have to agree on a specific format not specified here anyway, they can go off-spec anyway.

Comment (kbrockhoff) [reply specification/data-http.md]: Good point. I changed to header value only.

File: specification/data-http.md [hunk 2]
```diff
@@ -176,10 +176,12 @@ If the route cannot be determined, the `name` attribute MUST be set as defined i
 | `http.server_name` | The primary server name of the matched virtual host. This should be obtained via configuration. If no such configuration can be obtained, this attribute MUST NOT be set ( `net.host.name` should be used instead). | [1] |
 | `http.route` | The matched route (path template). (TODO: Define whether to prepend application root) E.g. `"/users/:userID?"`. | No |
 | `http.client_ip` | The IP address of the original client behind all proxies, if known (e.g. from [X-Forwarded-For][]). Note that this is not necessarily the same as `net.peer.ip`, which would identify the network-level peer, which may be a proxy. | No |
+| `http.user_agent` | Value of the HTTP [User-Agent] header sent by the client. | No |
```
Comment (SergeyKanzhelev) [inline specification/data-http.md]:
```suggestion
| `http.user_agent` | Value of the HTTP [User-Agent][] header sent by the client. | No |
```
make link work

Comment (SergeyKanzhelev) [reply specification/data-http.md]: hm... it seems to be working even without this. I wonder why this syntax is used above...

**Pull request merged**